histogram/histogram.go

```diff
@@ -75,7 +78,60 @@ func Hist(bins int, input []float64) Histogram {
 		Max:     maxC,
 		Count:   len(input),
 		Buckets: buckets,
+	}, nil
+}
+
+// HistByCustomRange creates an histogram by custom range, like elasticsearch data histogram query(left closed)
+func HistByCustomRange(min, max, interval float64, input []float64) (Histogram, error) {
+	if len(input) == 0 || interval == 0 {
+		return Histogram{}, nil
+	}
+
+	sort.Sort(sort.Float64Slice(input))
+
+	if min > input[len(input)-1] || max < input[0] {
+		return Histogram{}, nil
+	}
+
+	// input data equals range min and max
+	if min == max {
+		return Histogram{
+			Min:     len(input),
+			Max:     len(input),
+			Count:   len(input),
+			Buckets: []Bucket{{Count: len(input), Min: min, Max: max}},
+		}, nil
+	}
+
+	bins := cast.ToInt((max - min) / interval)
```

Comment:
Please change this code to do the cast manually, so that a whole new package doesn't need to be added to this project.